x-pack/plugins/endpoint/public/applications/endpoint/store/managing/reducer.ts

```diff
@@ -37,18 +39,24 @@ export const managementListReducer: Reducer<ManagementListState, AppAction> = (
       pageIndex,
       loading: false,
     };
-  }
-
-  if (action.type === 'userExitedManagementList') {
+  } else if (action.type === 'serverReturnedManagementDetails') {
+    return {
+      ...state,
+      details: action.payload,
+    };
+  } else if (action.type === 'userExitedManagementList') {
     return initialState();
-  }
-
-  if (action.type === 'userPaginatedManagementList') {
+  } else if (action.type === 'userPaginatedManagementList') {
     return {
       ...state,
       ...action.payload,
       loading: true,
     };
+  } else if (action.type === 'userChangedUrl') {
```

Comment:
@oatkiller @paul-tavares Are we ok with copying and pasting this in all of our reducers?

Reply:
@peluja1012 I don't like it either.
Personally, I think we should capture the route info. once and have a common (`lib`) method to retrieve/access it from anywhere. I think that would keep the middleware/reducer still "pure" (not force it to be aware of global state). 

Example:

1. `view/route_capture.tsx` could store the current location information locally in that module
2. `view/route_capture.tsx` could export a function (call it `getCurrentLocation(): EndpointAppLocation`) that we can use from anywhere (in or outside of react components) to gain access to the current location.

Just a example of how it could be done - probably not in the `view/` but in some `/lib/...`. With this in place, we could then have a generic `isOnPage(pageId | pageRoute)` type of getter (or multiples - one per page `isOnPolicyListPage()`, `isOnManagementPage()`, etc...

Any other ideas/suggestions?  Lets agree on an implementation design approach and then we can move forward on getting it through